### PR TITLE
fix(nimqml): Fixing memory leak - qmodelindex and char*

### PR DESCRIPTION
**Pull request by alexjba**
This PR is fixing two memory leaks identified in nimqml:
1. Release char pointers received from DOtherSide and used temporarily
2. Fix qmodelindex finalizer

Part of https://github.com/status-im/status-desktop/issues/9558

### CODE REVIEW

File: src/nimqml/private/qmodelindex.nim [hunk 1]
```diff
@@ -8,7 +8,7 @@ proc setup(self: QModelIndex, other: DosQModelIndex, takeOwnership: Ownership) =
 
 proc delete*(self: QModelIndex) =
   ## Delete the given QModelIndex
-  if not self.vptr.isNil:
+  if self.vptr.isNil:
```
Comment (osmaczko) [inline src/nimqml/private/qmodelindex.nim:11]: This looks like a massive leak, given a lot of model indexes are instantiated in our models.

**Review by osmaczko — APPROVED** [review]
Good catches. It would be nice if you could elaborate on how you identified these leaks. Simply valgrind?

**Review by jrainville — APPROVED** [review]

**Review by caybro — APPROVED** [review]

**Comment by alexjba** [comment]
@osmaczko I'm using [Instruments](https://en.wikipedia.org/wiki/Instruments_(software)) on macOS. Works pretty well with nim

File: src/nimqml/private/qmodelindex.nim (hunk 1 above)
Comment (saledjenic) [reply src/nimqml/private/qmodelindex.nim:11]: Nobody has ever check the repo we forked from.
It's still there: https://github.com/filcuc/nimqml/blob/master/src/nimqml/private/qmodelindex.nim#L9-L15

@alexjba you can open a pull request there as well.

Comment (alexjba) [reply src/nimqml/private/qmodelindex.nim:11]: @saledjenic good point. I will

**Pull request merged**